src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -246,7 +246,7 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                 case SyntaxKind.SingleVariableDesignation:
                     break;
                 case SyntaxKind.DiscardDesignation:
-                    return new BoundDeclarationPattern(node, null, boundDeclType, isVar, hasErrors);
+                    return new BoundDeclarationPattern(node, null, new BoundDiscardExpression(node, boundDeclType.Type), boundDeclType, isVar, hasErrors);
```

Comment:
Also restore the constructor that is, apparently, deleted in this PR. Place it into Constructors.cs

---
In reply to: [112778009](https://github.com/dotnet/roslyn/pull/18825#discussion_r112778009) [](ancestors = 112778009)